packages/vscode-extension/src/extension/main.ts

```diff
@@ -161,3 +167,32 @@ function startLanguageClient(context: vscode.ExtensionContext): LanguageClient {
   client.start();
   return client;
 }
+
+/**
+ * Watches the .pliplugin folder for changes to pgm_conf.json and proc_grps.json files.
+ * Sends a notification to the LS when changes are detected
+ */
+function watchPlipluginFolder(
+  client: LanguageClient,
+  workspaceFolder: string,
+  context: vscode.ExtensionContext,
+): void {
+  const plipluginPath = path.join(workspaceFolder, ".pliplugin");
+
+  if (!fs.existsSync(plipluginPath)) {
+    return;
+  }
+
+  const watcher = fs.watch(plipluginPath, (_eventType, filename) => {
+    if (filename === "pgm_conf.json" || filename === "proc_grps.json") {
+      client.sendNotification(WorkspaceDidChangePlipluginConfigNotification);
+    }
+  });
+
+  watcher.on("error", (error) => {
+    console.error("Error watching .pliplugin folder:", error);
+    watcher.close();
+  });
```

Comment:
Suggestion: This breaks pretty easily, for example if the `.pliplugin` folder doesn't exist during startup of the extension but is only added in later.

Please use the file watching options provided by VSCode, i.e. `vscode.workspace.createFileSystemWatcher`, which are much more robust. An alternative might be to instruct the language client to watch the folder, that way we don't actually need to base our implementation on the exact language client. See how [Langium is doing this](https://github.com/eclipse-langium/langium/blob/b9cd5299af415cab6045ad0a79e05526cdc62bce/packages/langium/src/lsp/document-update-handler.ts#L86-L120).

Reply:
Good point, just patched both those up (folder case + watcher abstraction), thanks for looking that over again. Also had no idea about the watcher abstraction under the vscode extension api, that's _much_ better in comparison.

Didn't opt to go the full route to use the LS to check client capabilities for dynamic registration, but that's good to know as well. I'm assuming we can set that up later as an improvement.